crypto/dsa/dsa_ossl.c

```diff
@@ -163,6 +166,12 @@ static int dsa_sign_setup(DSA *dsa, BN_CTX *ctx_in,
     } else
         ctx = ctx_in;
 
+    /* Preallocate space */
+    if (!BN_set_bit(k, BN_num_bits(dsa->q))
+        || !BN_set_bit(l, BN_num_bits(dsa->q))
+        || !BN_set_bit(m, BN_num_bits(dsa->q)))
```

Comment:
Here is interesting case of kind of "blindness" ;-) Why do you call BN_num_bits(dsa->q) multiple times? You kind of expect that compiler would figure it out, but it won't, would it? The function is opaque in this context and compiler will dutifully call it multiple times. [Note that there is another call later...]

Reply:
And same applies naturally to EC case...

Reply:
On second reading I'd like to replace "you" with "one" in above remark. It's wasn't my intention to address specifically Paul there. I rather meant that it's the kind of stuff we *all* tend to be kind of "blind" to...

[I have to apologize again, as I'm signing off for today...]

Reply:
I didn't take it personally.  I think everyone has written similar inefficient code.

Anyway, changed to cache the size.